src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -6,26 +6,22 @@
 public class ArrayListProductDao implements ProductDao {
 
     private List<Product> products;
-    private static ArrayListProductDao instance;
 
     private ArrayListProductDao() {
         products = new ArrayList<>();
     }
 
 
     public static ArrayListProductDao getInstance() {
-        if (instance == null) {
-            instance = new ArrayListProductDao();
-        }
-        return instance;
+        return ProductDaoHolder.productDao;
     }
 
 
     public List<Product> getAllProducts() {
         return products;
     }
 
-    protected void setProduct(Product product) {
+    public void setProduct(Product product) {
```

Comment:
Remove this method and write a protected setter method for product list to be used in test class